tests/functional/z_cluster/test_performance_profile_validation.py

```diff
@@ -0,0 +1,162 @@
+    Pod,
+)
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.ocs.resources.storage_cluster import StorageCluster
+
+
+log = logging.getLogger(__name__)
+
+
+@brown_squad
+@tier4a
+@skipif_external_mode
+@skipif_managed_service
+@skipif_ocs_version("<4.15")
+@pytest.mark.polarion_id("OCS-5645")
+@pytest.mark.polarion_id("OCS-5646")
+@pytest.mark.polarion_id("OCS-5656")
+@pytest.mark.polarion_id("OCS-5657")
+class TestProfileDefaultValuesCheck(ManageTest):
+    def test_validate_cluster_resource_profile(self):
+        """
+        Testcase to validate osd, mgr, mon, mds and rgw pod memory and cpu values
+        are matching with the predefined set of values post profile updation
+
+        """
+        pv_pod_obj = []
+        log.info("Obtaining the performance profile values from the cluster")
+        storage_cluster_name = config.ENV_DATA["storage_cluster_name"]
+        storage_cluster = StorageCluster(
+            resource_name=storage_cluster_name,
+            namespace=config.ENV_DATA["cluster_namespace"],
+        )
+        performance_profile = storage_cluster.data["spec"]["resourceProfile"]
+        if performance_profile == constants.PERFORMANCE_PROFILE_LEAN:
+            expected_cpu_request_values = constants.LEAN_PROFILE_REQUEST_CPU_VALUES
+            expected_memory_request_values = (
+                constants.LEAN_PROFILE_REQUEST_MEMORY_VALUES
+            )
+            expected_cpu_limit_values = constants.LEAN_PROFILE_CPU_LIMIT_VALUES
+            expected_memory_limit_values = constants.LEAN_PROFILE_MEMORY_LIMIT_VALUES
+        elif performance_profile == constants.PERFORMANCE_PROFILE_BALANCED:
+            expected_cpu_request_values = constants.BALANCED_PROFILE_REQUEST_CPU_VALUES
+            expected_memory_request_values = (
+                constants.BALANCED_PROFILE_REQUEST_MEMORY_VALUES
+            )
+            expected_cpu_limit_values = constants.BALANCED_PROFILE_CPU_LIMIT_VALUES
+            expected_memory_limit_values = (
+                constants.BALANCED_PROFILE_MEMORY_LIMIT_VALUES
+            )
+        elif performance_profile == constants.PERFORMANCE_PROFILE_PERFORMANCE:
+            expected_cpu_request_values = (
+                constants.PERFORMANCE_PROFILE_REQUEST_CPU_VALUES
+            )
+            expected_memory_request_values = (
+                constants.PERFORMANCE_PROFILE_REQUESt_LIMIT_VALUES
+            )
+            expected_cpu_limit_values = constants.PERFORMANCE_PROFILE_CPU_LIMIT_VALUES
+            expected_memory_limit_values = (
+                constants.PERFORMANCE_PROFILE_MEMORY_LIMIT_VALUES
+            )
+        else:
+            log.error("Does not match any performance profiles")
+
+        log.info(performance_profile)
+
+        label_selector = list(expected_cpu_limit_values.keys())
+
+        for label in label_selector:
+            for pod in get_pods_having_label(
+                label=label, namespace=config.ENV_DATA["cluster_namespace"]
```

Comment:
The logic does not seem to work correctly. The `label_selector` is the pod values: 'mgr', mon', 'osd', etc.
But the `label` values are different, as you can see here: https://github.com/pintojoy/ocs-ci/blob/ca45897ca2d0e41f1f51347950d45951fea79caf/ocs_ci/ocs/constants.py#L381. For example: `app=rook-ceph-mon`.

Reply:
Actually, I checked it in my local, and this will also work.